py/src/braintrust/test_framework.py

```diff
@@ -280,3 +282,106 @@ def simple_scorer(input, output, expected):
     # Most importantly: verify that no logs were sent (should be empty)
     logs = with_memory_logger.pop()
     assert len(logs) == 0
+
+
+@pytest.mark.asyncio
+async def test_hooks_tags_append(with_memory_logger, with_simulate_login, simple_scorer):
+    """ Test that hooks.tags can be appended to and logged. """
+
+    initial_tags = ["cookies n cream"]
+    appended_tags = ["chocolate", "vanilla", "strawberry"]
+    expected_tags = ["cookies n cream", "chocolate", "vanilla", "strawberry"]
+
+    def task_with_hooks(input, hooks):
+        for x in appended_tags:
+            hooks.tags.append(x)
+        return input
+
+    evaluator = Evaluator(
+        project_name=__name__,
+        eval_name=__name__,
+        data=[EvalCase(input="hello", expected="hello world", tags=initial_tags)],
+        task=task_with_hooks,
+        scores=[simple_scorer],
+        experiment_name=__name__,
+        metadata=None,
+        summarize_scores=False,
+    )
+    exp = init_test_exp(__name__)
+    result = await run_evaluator(experiment=exp, evaluator=evaluator, position=None, filters=[])
+    assert result.results[0].tags == expected_tags
+
+    logs = with_memory_logger.pop()
+    assert len(logs) == 3
+
+    # assert root span contains tags
+    root_span = [log for log in logs if not log["span_parents"]]
+    assert len(root_span) == 1
+    assert root_span[0].get("tags") == expected_tags
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(("tags", "expected_tags"), [([], None), (["chocolate", "vanilla", "strawberry"], ["chocolate", "vanilla", "strawberry"])])
+async def test_hooks_tags_list(with_memory_logger, with_simulate_login, simple_scorer, tags, expected_tags):
+    """ Test that hooks.tags can be set to a list. """
+
+    def task_with_hooks(input, hooks):
+        hooks.tags = tags
+        return input
+
+    evaluator = Evaluator(
+        project_name=__name__,
+        eval_name=__name__,
+        data=[EvalCase(input="hello", expected="hello world")],
```

Comment:
should we add a case for when `EvalCase` does not have initial tags AND the task does not touch `hook.tags`? that would express the behavior in my comment above and we can verify it is the same as it was before introducing the tags hook